python_ta/cfg/visitor.py

```diff
@@ -435,3 +441,16 @@ def _get_raise_exc(node: nodes.Raise) -> str:
         return f"{nodes.Raise.__name__} {next(exceptions).name}"
     except StopIteration:
         return nodes.Raise.__name__
+
+
+def _get_preconditions_node(func: nodes.FunctionDef) -> Optional[nodes.BoolOp]:
+    """A helper method that takes in a function definition node, retrieves its preconditions, and then parses them
+    into a AST node representing all the preconditions combined in an and statement. Returns None if there are no
+    preconditions."""
+    assertions = parse_assertions(func)
+    if not assertions:
+        return None
+    precondition_string = " and ".join(assertions)
+    condition, *_ = parse(precondition_string).nodes_of_class(nodes.Expr)
```

Comment:
Two comments here:

1. You should be able to use `extract_node` to obtain the minimal expression node, rather than using `nodes_of_class`
2. More importantly, this will fail if users specify non-Python preconditions (e.g. if there's a typo, or an English precondition). You should add test cases for this.

    In terms of expected behaviour, the easier approach is to return `None` in this case. The more complex approach is to filter out the non-Python preconditions and only join together the Python preconditions. I'll leave it to you to decide which approach to take, based on how much time you'd like to commit to this issue. 😃